lib/logstash/inputs/log4j.rb

```diff
@@ -8,6 +8,78 @@
 require "timeout"
 require 'logstash-input-log4j_jars'
 
+# DEPRECATION NOTICE
+# ==================
+#
+# NOTE: This plugin is deprecated. It is recommended that you use filebeat to collect logs from log4j.
+#
+# The following section is a guide for how to migrate from SocketAppender to use filebeat.
+#
+# To migrate away from log4j SocketAppender to using filebeat, you will need to make 3 changes:
+#
+# 1) Configure your log4j.properties (in your app) to write to a local file.
+# 2) Install and configure filebeat to collect those logs and ship them to Logstash
+# 3) Configure Logstash to use the beats input.
+#
+# Configuring log4j for writing to local files
+# --------------------------------------------
+# 
+# In your log4j.properties file, remove SocketAppender and replace it with RollingFileAppender. 
+#
+# For example, you can use the following log4j.properties configuration to write daily log files.
+#
+#     # Your app's log4j.properties (log4j 1.2 only)
+#     log4j.rootLogger=daily
+#     log4j.appender.daily=org.apache.log4j.rolling.RollingFileAppender
```

Comment:
RollingFileAppender in log4j v1 doesn't have any way to describe "Keep at most 7 old log files" so I don't know if this is the best recommendation. Users will need to configure log file cleanup externally.